tests/rules/unqualified-attributes.js

```diff
@@ -13,6 +13,16 @@
             Assert.areEqual("Unqualified attribute selectors are known to be slow.", result.messages[0].message);
         },
 
+        "A selector that contains a class is considered qualified": function(){
+            var result = CSSLint.verify("p [href].fancy[type=text] { font-size: 10px; }", {"unqualified-attributes": 1 });
+            Assert.areEqual(0, result.messages.length);
+        },
+
+        "A selector that contains an ID is considered qualified": function(){
```

Comment:
To match the bit below, I'd change this to: 

```
Using an attribute selector qualified by an ID should not result in a warning
```

I'd also move it to be nearer to line 38 test since they go together.